lib/savon/response.rb

```diff
@@ -69,7 +69,9 @@ def xpath(path, namespaces = nil)
 
     def find(*path)
       envelope = nori.find(hash, 'Envelope')
-      raise_invalid_response_error! unless envelope.is_a?(Hash)
+      unless envelope.is_a?(Hash)
+        raise_invalid_response_error! || envelope = {}
```

Comment:
this takes a dependency on `raise_invalid_response_error!` returning something falsey when `@globals[:raise_errors]` is enabled, or I think this might break. Is there a reason `envelope = {}` couldnt be on the following line?:

```
raise_invalid_response_error!
envelope = {}
```

I think this reveals that doing no-op in a method named `raise_invalid_response_error!` makes this code somewhat harder to follow

Reply:
Yeah, it could be on multiple lines. And if we have global flag that disables raising exceptions in favor of returning nil, then of course `raise_invalid_response_error! ` could be a no-op.

The other possibility is that this global flag is kinda "half baked" idea. Like we can't fully support it?